### PR TITLE
8332110: [macos] jpackage tries to sign added files without the --mac-sign option

### PR DESCRIPTION
**Pull request by sashamatveev**
This issue is reproducible with and without `--mac-sign`. jpackage will "_ad-hoc_" sign application bundle when `--mac-sign` is not specified by using pseudo-identity "_-_". This is why jpackage tries to sign added files and this is expected behavior by jpackage. "codesign" fails since added content made application bundle structure invalid. There is nothing we can do on jpackage side to sign such invalid bundles. As proposed solution we will output possible reason for "codesign" failure if it fails and `--app-content` was specified and possible solution. Proposed message: "One of the possible reason for "codesign" failure is additional content provided via "--app-content", which made application bundle structure invalid. Make sure to provide additional content in a way it will not break application bundle structure, otherwise add additional content as post-processing step."

Example:
Lets assume we have "ReadMe" folder with "ReadMe.txt" file in it.
1) jpackage --type app-image -n Test --app-content ReadMe/ReadMe.txt ...
"codesign" will fail with "In subcomponent: Test.app/Contents/ReadMe.txt". This is expected and "ReadMe.txt" placed in "Test.app/Contents" which is also expected.
2) jpackage --type app-image -n Test --app-content ReadMe ...
Works and "ReadMe.txt" will be placed under "Test.app/Contents/ReadMe".

Sample output before fix:
```
Error: "codesign" failed with following output:
Test.app: replacing existing signature
Test.app: code object is not signed at all
In subcomponent: Test.app/Contents/ReadMe.txt
```

Sample output after fix:
```
"codesign" failed and additional application content was supplied via the "--app-content" parameter. Probably the additional content broke the integrity of the application bundle and caused the failure. Ensure content supplied via the "--app-content" parameter does not break the integrity of the application bundle, or add it in the post-processing step.
Error: "codesign" failed with following output:
Test.app: replacing existing signature
Test.app: code object is not signed at all
In subcomponent: Test.app/Contents/ReadMe.txt
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8332110](https://bugs.openjdk.org/browse/JDK-8332110): [macos] jpackage tries to sign added files without the --mac-sign option (**Bug** - P3)

### Reviewers
 * [Alexey Semenyuk](https://openjdk.org/census#asemenyuk) (@alexeysemenyukoracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19377/head:pull/19377` \
`$ git checkout pull/19377`

Update a local copy of the PR: \
`$ git checkout pull/19377` \
`$ git pull https://git.openjdk.org/jdk.git pull/19377/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19377`

View PR using the GUI difftool: \
`$ git pr show -t 19377`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19377.diff">https://git.openjdk.org/jdk/pull/19377.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19377#issuecomment-2128306384)

### CODE REVIEW

**Comment by sashamatveev** [comment]
@alexeysemenyukoracle please review

**Comment by alexeysemenyukoracle** [comment]
How about this wording for the message:

"codesign" failed and additional application content was supplied via the "--app-content" parameter. Probably the additional content broke the integrity of the application bundle and caused the failure. Ensure content supplied via the "--app-content" parameter does not break the integrity of the application bundle, or add it in the post-processing step.

**Comment by sashamatveev** [comment]
8332110: jpackage tries to sign added files without the --mac-sign option [v2]
 - Updated error message as suggested.

**Review by alexeysemenyukoracle — APPROVED** [review]

File: test/jdk/tools/jpackage/macosx/SigningOptionsTest.java [hunk 1]
```diff
@@ -87,6 +90,11 @@ public static Collection input() {
                                  "--mac-installer-sign-identity", "test-identity"},
                     new String[]{"--type"},
                     "Option [--mac-installer-sign-identity] is not valid with type"},
+            // --mac-installer-sign-identity and --type dmg
+            {"Hello",
+                    new String[]{"--app-content", TEST_DUKE},
+                    null,
+                    "\"codesign\" failure is additional content provided via \"--app-content\""},
```
[the remark this answers is not in the document]
Comment (sashamatveev) [reply test/jdk/tools/jpackage/macosx/SigningOptionsTest.java]: It was part of old message. Latest test version has latest message: "\"codesign\" failed and additional application content was supplied via the \"--app-content\" parameter."

**Comment by nlisker** [comment]
> > For your example. This almost seems like an Apple bug if you can add a directory to the Contents directory but not a file?
> 
> Not sure if it is an Apple bug.

Can this be reported to Apple somehow?

**Comment by sashamatveev** [comment]
> > > For your example. This almost seems like an Apple bug if you can add a directory to the Contents directory but not a file?
> > 
> > 
> > Not sure if it is an Apple bug.
> 
> Can this be reported to Apple somehow?

I do not think that it is an Apple bug. If you look at an Apple documentation [1] about "Placing content in a bundle" you will see highlighted important message that you should not put content in the wrong location.

[1] https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle?language=objc

**Comment by nlisker** [comment]
I see, but it doesn't say where to put license files, which are usually in the root. Do you know where these belong?

**Comment by sashamatveev** [comment]
> I see, but it doesn't say where to put license files, which are usually in the root. Do you know where these belong?

No idea.
